### ISSUE TITLE
Deploy Script doesn't start deploy

### ISSUE THREAD

**Issue by miguelduque** [microsoft/fhir-server-samples#113]
Hello,

After logging in an account where I'm admin and have all permissions, I'm trying to deploy the managed Azure API for FHIR by running the following command:
`.\Create-FhirServerSamplesAuthConfig.ps1 -EnvironmentName duphill2 -UsePaaS $true`

Unfortunately, the only outcome is:
- Creation of the Resource group "duphill2"
- Creation of the Key vault "duphill2-ts" inside the resource group

The feedback on the powershell console is:
```
C:\new\fhir-server-samples\deploy\scripts [master ≡]> .\Create-FhirServerSamplesAuthConfig.ps1 -EnvironmentName duphill2 -UsePaaS $false

duphill2
FhirServer PS module is loaded
Creating keyvault with the name duphill2-ts
WARNING: The provided information does not map to an AD object id.
WARNING: Access policy is not set. No user or application have access permission to use this vault. This can happen if the vault was created by a
service principal. Please use Set-AzKeyVaultAccessPolicy to set access policies.
Current context is user: ...@....
Current context is user: ...@....
Adding permission to keyvault for ....
Ensuring API application exists
Checking if UserPrincipalName exists
User not, will create.
```

These are my PowerShell modules:
```
Version    Name                                Repository           Description
-------    ----                                ----------           -----------
5.2.0      Az                                  PSGallery            Microsoft Azure PowerShell - Cmdlets to manage resources in Azure. This module i...
2.2.2      Az.Accounts                         PSGallery            Microsoft Azure PowerShell - Accounts credential management cmdlets for Azure Re...
1.1.1      Az.Advisor                          PSGallery            Microsoft Azure PowerShell - Azure Advisor Cmdlets for Advisor in Windows PowerS...
2.0.1      Az.Aks                              PSGallery            Microsoft Azure PowerShell - Azure managed Kubernetes cmdlets for Windows PowerS...
1.1.4      Az.AnalysisServices                 PSGallery            Microsoft Azure PowerShell - Analysis Services cmdlets for Windows PowerShell an...
2.2.0      Az.ApiManagement                    PSGallery            Microsoft Azure PowerShell - Api Management service cmdlets for Azure Resource M...
1.0.0      Az.AppConfiguration                 PSGallery            Microsoft Azure PowerShell: AppConfiguration cmdlets
1.1.0      Az.ApplicationInsights              PSGallery            Microsoft Azure PowerShell - Application Insights management cmdlets for Azure R...
1.4.0      Az.Automation                       PSGallery            Microsoft Azure PowerShell - Automation service cmdlets for Azure Resource Manag...
3.1.0      Az.Batch                            PSGallery            Microsoft Azure PowerShell - Batch service cmdlets for Azure Resource Manager in...
2.0.0      Az.Billing                          PSGallery            Microsoft Azure PowerShell - Billing service cmdlets for Azure Resource Manager ...
1.6.0      Az.Cdn                              PSGallery            Microsoft Azure PowerShell - CDN service cmdlets for Azure Resource Manager in W...
1.8.0      Az.CognitiveServices                PSGallery            Microsoft Azure PowerShell - Cognitive Services management cmdlets for Azure Res...
4.7.0      Az.Compute                          PSGallery            Microsoft Azure PowerShell - Compute service cmdlets for Azure Resource Manager ...
1.0.3      Az.ContainerInstance                PSGallery            Microsoft Azure PowerShell - Container Instance cmdlets for Azure Resource Manag...
2.1.0      Az.ContainerRegistry                PSGallery            Microsoft Azure PowerShell - Container Registry service cmdlets for Azure Resour...
1.1.0      Az.DataBoxEdge                      PSGallery            Microsoft Azure PowerShell - DataBoxEdge service cmdlets for Azure Resource Mana...
1.0.1      Az.Databricks                       PSGallery            Microsoft Azure PowerShell: Databricks cmdlets
1.11.2     Az.DataFactory                      PSGallery            Microsoft Azure PowerShell - Data Factory service cmdlets for Azure Resource Man...
1.0.2      Az.DataLakeAnalytics                PSGallery            Microsoft Azure PowerShell - Data Lake Analytics in Windows PowerShell and Power...
1.3.0      Az.DataLakeStore                    PSGallery            Microsoft Azure PowerShell - Azure Data Lake Store cmdlets in Windows PowerShell...
1.0.0      Az.DataShare                        PSGallery            Microsoft Azure PowerShell - DataShare service cmdlets for Azure Resource Manage...
1.1.0      Az.DeploymentManager                PSGallery            PowerShell .Net Core Microsoft Azure PowerShell - Deployment Manager cmdlets for...
2.0.1      Az.DesktopVirtualization            PSGallery            Microsoft Azure PowerShell: DesktopVirtualization cmdlets
1.0.2      Az.DevTestLabs                      PSGallery            Microsoft Azure PowerShell - DevTest Labs service cmdlets for Azure Resource Man...
1.1.2      Az.Dns                              PSGallery            Microsoft Azure PowerShell - DNS service cmdlets for Azure Resource Manager in W...
1.3.0      Az.EventGrid                        PSGallery            Microsoft Azure PowerShell - Event Grid service cmdlets for Azure Resource Manag...
1.7.1      Az.EventHub                         PSGallery            Microsoft Azure PowerShell - Event Hubs service cmdlets for Azure Resource Manag...
1.6.1      Az.FrontDoor                        PSGallery            Microsoft Azure PowerShell - Front Door service cmdlets for Azure Resource Manag...
2.0.0      Az.Functions                        PSGallery            Microsoft Azure PowerShell - Azure Functions service cmdlets for Azure Resource ...
4.1.0      Az.HDInsight                        PSGallery            Microsoft Azure PowerShell - HDInsight service cmdlets for Azure Resource Manage...
1.2.0      Az.HealthcareApis                   PSGallery            Microsoft Azure PowerShell - HealthcareApis management cmdlets for Azure Resourc...
2.7.1      Az.IotHub                           PSGallery            Microsoft Azure PowerShell - IoT Hub service cmdlets for Azure Resource Manager ...
3.2.0      Az.KeyVault                         PSGallery            Microsoft Azure PowerShell - Key Vault service cmdlets for Azure Resource Manage...
1.0.0      Az.Kusto                            PSGallery            Microsoft Azure PowerShell: Kusto cmdlets
1.4.0      Az.LogicApp                         PSGallery            Microsoft Azure PowerShell - Logic Apps cmdlets for Azure Resource Manager in Wi...
1.1.3      Az.MachineLearning                  PSGallery            Microsoft Azure PowerShell - Machine Learning Web Services cmdlets for Azure Res...
1.1.0      Az.Maintenance                      PSGallery            Microsoft Azure PowerShell - Maintenance cmdlets for Azure Resource Manager in W...
2.0.0      Az.ManagedServices                  PSGallery            Microsoft Azure PowerShell - ManagedServices cmdlets for Azure Resource Manager
1.0.2      Az.MarketplaceOrdering              PSGallery            Microsoft Azure PowerShell - Marketplace Ordering agreements service cmdlets for...
1.1.1      Az.Media                            PSGallery            Microsoft Azure PowerShell - Media service cmdlets for Azure Resource Manager in...
2.3.0      Az.Monitor                          PSGallery            Microsoft Azure PowerShell - Monitor service cmdlets for Azure Resource Manager ...
4.3.0      Az.Network                          PSGallery            Microsoft Azure PowerShell - Networking service cmdlets for Azure Resource Manag...
1.1.1      Az.NotificationHubs                 PSGallery            Microsoft Azure PowerShell - Notification Hubs cmdlets for Azure Resource Manage...
2.3.0      Az.OperationalInsights              PSGallery            Microsoft Azure PowerShell - Operational Insights service cmdlets for Azure Reso...
1.3.1      Az.PolicyInsights                   PSGallery            Microsoft Azure PowerShell - Azure Policy Insights cmdlets for Windows PowerShel...
1.1.2      Az.PowerBIEmbedded                  PSGallery            Microsoft Azure PowerShell - Power BI Embedded service management cmdlets for Az...
1.0.3      Az.PrivateDns                       PSGallery            Microsoft Azure PowerShell - Private DNS service cmdlets for Azure Resource Mana...
3.1.0      Az.RecoveryServices                 PSGallery            Microsoft Azure PowerShell - Recovery Services cmdlets for Azure Resource Manage...
1.4.0      Az.RedisCache                       PSGallery            Microsoft Azure PowerShell - Redis Cache service cmdlets for Azure Resource Mana...
1.0.3      Az.Relay                            PSGallery            Microsoft Azure PowerShell - Relay service cmdlets for Azure Resource Manager in...
3.1.0      Az.Resources                        PSGallery            Microsoft Azure PowerShell - Azure Resource Manager and Active Directory cmdlets...
1.4.1      Az.ServiceBus                       PSGallery            Microsoft Azure PowerShell - Service Bus service cmdlets for Azure Resource Mana...
2.2.1      Az.ServiceFabric                    PSGallery            Microsoft Azure PowerShell - Service Fabric cmdlets for Azure Resource Manager i...
1.2.0      Az.SignalR                          PSGallery            Microsoft Azure PowerShell - Azure SignalR service commands for Windows PowerShe...
2.13.0     Az.Sql                              PSGallery            Microsoft Azure PowerShell - SQL service cmdlets for Azure Resource Manager in W...
1.1.0      Az.SqlVirtualMachine                PSGallery            Microsoft Azure PowerShell - SqlVirtualMachine service cmdlets for Azure Resourc...
3.1.0      Az.Storage                          PSGallery            Microsoft Azure PowerShell - Storage service data plane and management cmdlets f...
1.4.0      Az.StorageSync                      PSGallery            Microsoft Azure PowerShell - Storage Sync cmdlets in Windows PowerShell and Powe...
1.0.1      Az.StreamAnalytics                  PSGallery            Microsoft Azure PowerShell - Stream Analytics service cmdlets for Azure Resource...
1.0.0      Az.Support                          PSGallery            Microsoft Azure PowerShell - Azure Support cmdlets for Azure Resource Manager in...
1.0.4      Az.TrafficManager                   PSGallery            Microsoft Azure PowerShell - Traffic Manager service cmdlets for Azure Resource ...
2.1.1      Az.Websites                         PSGallery            Microsoft Azure PowerShell - App Service (Web Apps) service cmdlets for Azure Re...
2.0.2.128  AzureAD                             PSGallery            Azure Active Directory V2 General Availability Module....
0.7.3      posh-git                            PSGallery            Provides prompt with Git status summary information and tab completion for Git c...
```

Am I missing something?

**Comment by leecrossley** [comment]
I assume you created a test tenant but didn't add your user to it? try the default tenant (if you have permission) or add your user. Well done for getting that far! 👍

**Comment by miguelduque** [comment]
So I ended up trying to deploy the services without the sandbox script.

- I manually deployed the Azure API for FHIR
- I used the azuredeploy-aci-importer.json (https://github.com/microsoft/fhir-server-samples/blob/master/deploy/templates/azuredeploy-aci-importer.json) to deploy the importer

_At this point, I manage to import the Synthea files and get Patient information throught the FHIR rest endpoints (using Postman)_

- Then I used the azuredeploy-fhirdashboard-js.json (https://github.com/microsoft/fhir-server-samples/blob/master/deploy/templates/azuredeploy-fhirdashboard-js.json) to deploy a dashboard on top of the API

The dashboard deployed successfully, however, I'm stuck with this error when logging in the dashboard app:
```
AADSTS650057: Invalid resource. 
The client has requested access to a resource which is not listed in the requested permissions in the client's application registration. 
Client app ID: ...uuid... (dashboard-app-name). 
Resource value from request: https://resource-group-name.azurehealthcareapis.com. 
Resource app ID: ...uuid... 
List of valid resources from app registration: 00000003-0000-0000-c000-000000000000.
```

**Comment by jeffbonasso-ambifi** [comment]
I am having same issue as Miguel deploying. The error I get from Deploy Script is...
The property 'SecretValueText' cannot be found on this object.

Miguel, I am new to Azure, when you say you used the JSON file to deploy, can you tell me the steps you use to deploy the JSON file manually?

**Comment by jeffbonasso-ambifi** [comment]
I ran the command from a local PowerShell command window on a PC. What was weird is the latest PowerShell 7 didn't work. I had to use the one that ships with Windows 10 which is PowerShell 5.1.

**Comment by ginalee-dotcom** [comment]
It seems like certain PowerShell modules still have not been fixed, which unfortunately is not something we can fix or expedite the fix. Please refer to README for the working versions of PowerShell for the deployment! Apologies for the inconvenience on the PowerShell version dependency - we are working on an alternative way of not using PowerShell.
Regarding `SecretValueText` error, `Az.KeyVault` has removed the attribute `SecretValueText` to avoid logging or displaying the Secret Value directly. I merged a PR that potentially fixes this error: #117
Hope this helps!
